ehsm_kms_service/router.js

```diff
@@ -0,0 +1,140 @@
+const {
+  ehsm_keySpec_t,
+  ehsm_keyorigin_t,
+  ehsm_napi_params_keys_list,
+} = require('./ehsm_kms_params.js')
+const { cryptographic_apis, enroll_apis } = require('./apis')
+const logger = require('./logger')
+const {
+  napi_result,
+  _result,
+  create_app_info,
+  save_cmk,
+} = require('./function')
+
+// napi params
+const _params_list = (params_keys, payload) => {
+  const res = []
+  for (const key of params_keys) {
+    res.push(payload[key])
+  }
+  return res
+}
+
+const cmk_db_query = (id) => {
+  return {
+    selector: {
+      _id: `cmk:${id}`,
+    },
+    fields: ['keyBlob', 'creator', 'expireTime'],
+    limit: 1,
+  }
+}
+
+async function search_cmk(appid, keyid, DB) {
+  const query = cmk_db_query(keyid)
+  const cmk = await DB.partitionedFind('cmk', query)
+  if (!cmk.docs[0].keyBlob) {
+    res.send(_result(400, 'keyid error'))
+    return false
+  }
+  if (appid != cmk.docs[0].creator) {
+    res.send(_result(400, 'appid error'))
+    return false
+  }
+  if (new Date().getTime() > cmk.docs[0].expireTime) {
+    res.send(_result(400, 'keyid expire'))
+    return
+  }
+  return cmk.docs[0].keyBlob
+}
+
+// all cryptographic_apis without CreateKey
+async function cryptographic_apis_with_cmk(
+  action,
+  res,
+  current_ehsm_napi_params_keys,
+  appid,
+  payload,
+  DB
+) {
+  try {
+    const { keyid, ukeyid } = payload
+    const current_payload = { ...payload }
+    if (keyid) {
+      const cmk_base64 = await search_cmk(appid, keyid, DB)

```

Comment:
search_cmk() returns base64 keyblob or key id ?

Reply:
it will be change